USWDS-Site: Add icon accessibility tests page

amyleadem opened the pull request:
# Summary
Added icon accessibility test page.

> [!Note]
> These tests for this component are all marked as conditional. This is because there are different methods for including an icon on a page, each with different levels of accessibility. This variability made the icon tests feel different from the other components in the system, and the team felt like there should be extra emphasis on testing icon implementation in each project. 
>
> The team drafted a sample explanation that we can include on the page, if desired. It is not included on the page yet, but we can work to include something like this now or as a quick follow-on:
>
> _Icons can be decorative or functional. How you check an icon for compliance depends on how it's used. This is why the the checks below are marked as conditional. Our icon library is accessible to 2.1 AA standards. Follow the checks below to determine if your use of an icon passes each success criteria._

> [!Important]
> We need to confirm the changelog dates before merge.

## Resources
- [Creating accessibility checklists process doc](https://docs.google.com/document/d/14WTCsdzgwpoz_Miztd66RmEGfwwSEUFCVN1DkSQITTE/edit) (Google docs 🔒)
- [Accessibility checklist spreadsheet](https://docs.google.com/spreadsheets/d/1fAJaObQHv7aIaeUpH4fTLmauhmugMtNCrbpZe4BJcPY/edit#gid=97481116) (Google sheets 🔒)

## Related issue
Closes #2535 

## Preview link
- [Icon - accessibility test page](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-icon-checklist/components/icon/accessibility-tests/)
- [Icon main page](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/al-icon-checklist/components/icon/)

## Testing and review
1. Confirm that both the main component page and the accessibility tests page has the correct compliance tag at the top.
1. Confirm that the main component page links to the accessibility tests page in the side navigation.
1. Confirm that the main component page links to the accessibility tests page in the accessibility section. 
    1. Confirm that this text references the correct component. For example "Test the icon in your own project." 
1. Check that accessibility tests page provides the correct counts for each test status type.
1. Confirm the test checklist summaries and data are accurate.  
1. Confirm no visual issues.
1. Confirm there is an appropriate changelog entry on both the main component page and the accessibility tests page.

amyleadem reviewed (commented):
@amycole501 @alex-hull @finekatie @sarah-sch 

I've added some comments about the content for these checklist items. Happy to meet up to discuss if that is simpler!

amyleadem commented on _data/changelogs/component-icon-accessibility.yml:
```diff
@@ -0,0 +1,9 @@
+  - date: NNNN-NN-NN
+    summary: Added accessibility tests page.
+    affectsGuidance: true
+    githubPr:
```
```suggestion
    githubPr: 2518
```

finekatie commented on _data/accessibility-tests/icon.yml:
```diff
@@ -0,0 +1,61 @@
+# Screen reader items
+  - summary: Standalone icons provide a text alternative for screen readers.
+    summary_additional: |
+      When you listen to an icon's description using assistive technology
+      and the icon receives focus,
+      standalone icons have descriptions for screen readers.
```
I think this works well. I support incorporating it.

amyleadem replied:
Updated in [4b983fc](https://github.com/uswds/uswds-site/pull/2518/commits/4b983fc1de22af0dc471a318c5e0c3415d2a53b3). Let me know if I should revert it,

finekatie commented on _data/accessibility-tests/icon.yml:
```diff
@@ -0,0 +1,61 @@
+    wcag_criterion: 1.4.11
+# Zoom/screen magnification items
+# Keyboard navigation items
+  - summary: Focus indicator is clearly visible on interactive icons.
```
Good question. I usually support plural, but I wonder if it might suggest multiple indicators. Thoughts, @sarah-sch ?

amyleadem replied:
Updated to plural in [4b983fc](https://github.com/uswds/uswds-site/pull/2518/commits/4b983fc1de22af0dc471a318c5e0c3415d2a53b3).

finekatie commented on _data/accessibility-tests/icon.yml:
```diff
@@ -0,0 +1,61 @@
+  - summary: Focus indicator is clearly visible on interactive icons.
+    summary_additional: |
+      When you use a keyboard to tab into an interactive icon and the icon receives focus,
+      you see a visible indicator around the icon.
```
Off the top of my head, I'm thinking sometimes we might use one over the other for the purposes of plain language and brevity. As in the case above, what we have is shorter and hopefully simpler than the updated version. The question becomes: which is easier for the user to understand? Is the current version not clear enough? If it isn't, then I'd go with the update.

amyleadem replied:
I'll leave it as is for now. In the future, it would be great to make statements like these consistent across components whenever possible, but it doesn't need to be now.

amyleadem commented on _data/accessibility-tests/icon.yml:
```diff
@@ -0,0 +1,61 @@
+      the icon's meaning, and the way it is announced, is the same across all pages.
+    test_status: conditional
+    status_details: This can only be tested in the context of your site.
+    test_type: screen_reader
```
I moved this to general in [4b983fc](https://github.com/uswds/uswds-site/pull/2518/commits/4b983fc1de22af0dc471a318c5e0c3415d2a53b3) because it seemed to apply to both visible and screen reader text.

amyleadem commented:
@sarah-sch @finekatie @amycole501 @alex-hull 
I have incorporated the changes discussed in our conversation yesterday and this is ready for review. I've added a couple of comments in purple to Column L of the [checklist spreadsheet](https://docs.google.com/spreadsheets/d/1fAJaObQHv7aIaeUpH4fTLmauhmugMtNCrbpZe4BJcPY/edit#gid=97481116) highlighting items I didn't include on the page at this time. Please review any open comments in the issue above and let me know if you need any changes.

amyleadem commented on _data/accessibility-tests/icon.yml:
```diff
@@ -0,0 +1,61 @@
+    status_details: This can only be tested in the context of your site.
+    test_type: screen_reader
+    version_tested:
+    wcag_criterion: 1.3.6
```
After discussion, I removed this test item from the page. There are a few reasons for: It is a AAA criterion that seems to require the user to look at and understand code. Additionally, WCAG does not provide clear guidance for what the expected markup for icons should be. The 1.1.1 rule appears to functionally cover the requirements here.

sarah-sch commented on _data/accessibility-tests/icon.yml:
```diff
@@ -0,0 +1,50 @@
+# General items
+  - summary: Color is not the only method used to convey icon information.
+    summary_additional: |
+      When you view an icon that represents an action and determine the action,
```
Suggest edit: "When you view an icon paired with, text, and never color alone, conveys the icon’s meaning."

amyleadem replied:
Updated in [abe962f](https://github.com/uswds/uswds-site/pull/2518/commits/abe962f72b0e14d17d23ecf9edbb4e406da5e570)

sarah-sch commented on _data/accessibility-tests/icon.yml:
```diff
@@ -0,0 +1,50 @@
+  - summary: Each icon has a consistent meaning across pages.
+    summary_additional: |
+      When you use a screen reader or view the page
+      and determine the purpose of the icon,
```
I suggest deleting "and determine the purpose of the page" here

amyleadem replied:
Updated in [abe962f](https://github.com/uswds/uswds-site/pull/2518/commits/abe962f72b0e14d17d23ecf9edbb4e406da5e570)

sarah-sch reviewed (changes requested):
Along with my in-line suggestions, please change the text above the table (at top of page) to read as follows: "The USWDS team did five tests based on WCAG 2.1 AA success criteria." (Spell out five instead of using a numeral because it's less than 10.)

amycole501 reviewed (approved):
I like Sarah's comments; the page looks good to me.

amyleadem commented:
@sarah-sch Your requested updates should now be live. Let me know if you need any more changes!

sarah-sch commented on _data/accessibility-tests/icon.yml:
```diff
@@ -0,0 +1,61 @@
+# Keyboard navigation items
+  - summary: Focus indicator is clearly visible on interactive icons.
+    summary_additional: |
+      When you use a keyboard to tab into an interactive icon and the icon receives focus,
```
I agree with this suggestion, Amy.

amyleadem replied:
Removed in [95fb48e](https://github.com/uswds/uswds-site/pull/2518/commits/95fb48e3aac630394828661e063eae9e63d4c6e4)

sarah-sch reviewed (approved):
Looks good to publish as an MVP after fed review

amyleadem commented on _data/accessibility-tests/icon.yml:
```diff
@@ -0,0 +1,61 @@
+  - summary: Color is not the only method used to convey icon information.
+    summary_additional: |
+      When you view an icon that represents an action and determine the action,
+      the text (not color alone) conveys the icon's meaning where necessary.
```
Marking this as resolved because it was addressed in [this comment](https://github.com/uswds/uswds-site/pull/2518#discussion_r1532633216) instead.

amyleadem commented:
@thisisdano @annepetersen This is ready for your review. FYI - I am able to get pa11y tests to pass locally, but they are still crashing here. Going to keep trying to get them to pass here 🏄

thisisdano commented:
Merging over crashing pa11y test
